xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -3059,6 +3049,30 @@ IAESound *CActiveAE::MakeSound(const std::string& file)
         delete sound;
         return NULL;
       }
+
+      if (sent != -1)
+      {
+        receive_frame_error = avcodec_receive_frame(dec_ctx, decoded_frame);
```

Comment:
I think this this method could be simplified a lot if it handled read and send in a loop, then receive in a separate one